app/scripts/directives/serviceBinding.js

```diff
@@ -20,23 +23,62 @@
   });
 
   function ServiceBinding(APIService,
+                          AuthorizationService,
+                          DataService,
+                          SecretsService,
                           ServiceInstancesService) {
     var ctrl = this;
     ctrl.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    ctrl.showParameterValues = false;
+
+    var context = {
+      namespace: ctrl.namespace
+    };
+
+    var updateParameterData = function() {
+      ctrl.parameterData = angular.copy(_.get(ctrl.binding, 'spec.parameters', {}));
+      if (AuthorizationService.canI('secrets', 'get', ctrl.namespace)) {
+        _.each(_.get(ctrl.binding, 'spec.parametersFrom'), function (parametersSource) {
+          DataService.get('secrets', _.get(parametersSource, 'secretKeyRef.name'), context).then(function (secret) {
+            try {
+              _.extend(ctrl.parameterData, SecretsService.decodeSecretData(secret.data).parameters);
```

Comment:
This is not always `parameters`. It is the `secretKeyRef.key`

Reply:
Surprised you don't need a `JSON.parse` here

Reply:
Yes. Thanks, I do need a JSON.parse here.